iml-sfa/src/main.rs

```diff
@@ -36,7 +36,7 @@ enum ImlSfaError {
 
 fn create_retry_endpoint_policy<E: Debug>(len: u32) -> impl RetryPolicy<E> + Send {
     move |k: u32, e| {
-        if k < len {
+        if k < len - 1 {
             RetryAction::RetryNow
         } else {
             RetryAction::ReturnError(e)
```

Comment:
```rust
if k >= len {
   RetryAction::ReturnError(e)
} else {
  RetryAction::RetryNow
}
```

:) Just an idea

Reply:
And you probably would not need a test for that.

Reply:
sure, makes sense

Reply:
actually, my unit test failed with this. Looking.

Reply:
ah right, it would have to be >= len - 1. I think i'm going to leave it as is.